nav2_bringup/urdf/turtlebot3_waffle.urdf

```diff
@@ -290,4 +290,164 @@
   </joint>
   <link name="camera_depth_optical_frame"/>
 
+  <gazebo reference="base_link">
```

Comment:
we have this defined in waffle.model https://github.com/ros-planning/navigation2/blob/c52b2b051b6d706cb47059df645a38b33fbaacfb/nav2_bringup/worlds/waffle.model

This is not used in this context

Reply:
I was using a different approach, we are using two files for the same.
 
 - robot_sdf (`waffle.model`). It's used to spawn the model
 - `turtlebot3_waffle` is hardcoded in the file and it's used to set the `robot_description` in the `robot_state_publisher`

we can use the urdf for both. I might consider this as a bug because you can configure the `robot_sdf` with a value but then the `robot_description` might publish other completely different robot model.

I can open a PR to fix this.

Reply:
> we can use the urdf for both

I don't disagree and its usually how I have done it in the past, but I don't want to mess with what aim broke or else I can't backport this PR to Humble / Iron, which I'd really like to. I'd like this to be purely additive instead of modifying of things that I would consider to be a breaking change